sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2966,6 +2966,17 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-26709: OptimizeMetadataOnlyQuery does not handle empty records correctly") {
```

Comment:
Also add a case that illustrates that `OptimizeMetadataOnlyQuery` can produce wrong results.

Reply:
Also, can we add a test case for hive tables?, e.g.,
```

scala> sql("CREATE TABLE t (id STRING) PARTITIONED BY (p STRING)")
scala> sql("INSERT INTO t VALUES ('1', '3')")
scala> sql("TRUNCATE TABLE t PARTITION (p = '3')")
scala> sql("SET spark.sql.optimizer.metadataOnly=true")
scala> spark.table("t").selectExpr("max(p)").show
+------+
|max(p)|
+------+
|     3|
+------+

scala> sql("SET spark.sql.optimizer.metadataOnly=false")
scala> spark.table("t").selectExpr("max(p)").show
+------+
|max(p)|
+------+
|  null|
+------+
```